server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateIntervalWrapper.java

```diff
@@ -300,58 +251,11 @@ private void setIntervalType(IntervalTypeEnum type) {
             intervalType = type;
             return;
         }
-
-        // interval() method
-        switch (type) {
-            case LEGACY_INTERVAL:
-                if (intervalType.equals(IntervalTypeEnum.CALENDAR) || intervalType.equals(IntervalTypeEnum.FIXED)) {
-                    throw new IllegalArgumentException("Cannot use [interval] with [fixed_interval] or [calendar_interval] " +
-                        "configuration options.");
-                }
-
-                // dateHistogramInterval() takes precedence over interval()
-                if (intervalType.equals(IntervalTypeEnum.LEGACY_DATE_HISTO) == false) {
-                    intervalType = IntervalTypeEnum.LEGACY_INTERVAL;
-                }
-                break;
-
-            case LEGACY_DATE_HISTO:
-                if (intervalType.equals(IntervalTypeEnum.CALENDAR) || intervalType.equals(IntervalTypeEnum.FIXED)) {
-                    throw new IllegalArgumentException("Cannot use [interval] with [fixed_interval] or [calendar_interval] " +
-                        "configuration options.");
-                }
-
-                // dateHistogramInterval() takes precedence over interval()
-                intervalType = IntervalTypeEnum.LEGACY_DATE_HISTO;
-                break;
-
-            case FIXED:
-                if (intervalType.equals(IntervalTypeEnum.LEGACY_INTERVAL) || intervalType.equals(IntervalTypeEnum.LEGACY_DATE_HISTO)) {
-                    throw new IllegalArgumentException("Cannot use [fixed_interval] with [interval] " +
-                        "configuration option.");
-                }
-                if (intervalType.equals(IntervalTypeEnum.CALENDAR)) {
-                    throw new IllegalArgumentException("Cannot use [fixed_interval] with [calendar_interval] " +
-                        "configuration option.");
-                }
-                intervalType = IntervalTypeEnum.FIXED;
-                break;
-
-            case CALENDAR:
-                if (intervalType.equals(IntervalTypeEnum.LEGACY_INTERVAL) || intervalType.equals(IntervalTypeEnum.LEGACY_DATE_HISTO)) {
-                    throw new IllegalArgumentException("Cannot use [calendar_interval] with [interval] " +
-                        "configuration option.");
-                }
-                if (intervalType.equals(IntervalTypeEnum.FIXED)) {
-                    throw new IllegalArgumentException("Cannot use [calendar_interval] with [fixed_interval] " +
-                        "configuration option.");
-                }
-                intervalType = IntervalTypeEnum.CALENDAR;
-                break;
-
-            default:
-                throw new IllegalStateException("Unknown interval type.");
+        if (type.isValid() == false || intervalType.isValid() == false) {
+            throw new IllegalStateException("Unknown interval type.");
```

Comment:
Nit: Should this be an `IllegalArgumentException` instead of a `IllegalStateException`.